rust/flexbuffers/src/internal_buffer.rs

```diff
@@ -0,0 +1,71 @@
+use std::fmt::Debug;
+use std::ops::{Deref, RangeBounds, Bound};
+
+/// The underlying buffer that is used by a flexbuffer Reader. 
+///
+/// This allows for custom buffer implementations so long as they can be
+/// viewed as a &[u8]. 
+pub trait InternalBuffer: Deref<Target = [u8]> + AsRef<[u8]> + Clone + Default + Debug {
+    type BufferString: Deref<Target = str> + AsRef<str> + Default + Debug;
```

Comment:
I went with `Deref` since it's a bit more direct. It also makes it so there's a smaller code impact (and the code looks cleaner) between this branch and master.